transform/binary-plugin/examples/openshift/openshift.go

```diff
@@ -6,82 +6,232 @@ import (
+			replaceJSON = fmt.Sprintf(
+				replaceOp,
+				replaceJSON,
+				nonInitialDelimiter(priorReplace),
+				i,
+				newSecret,
+			)
+			priorReplace = true
+		} else if fields.StripDefaultPullSecrets && isDefault(pullSecrets[i].Name) {
+			removeJSON = fmt.Sprintf(
+				removeOp,
+				removeJSON,
+				nonInitialDelimiter(priorRemove),
+				i,
+			)
+			priorRemove = true
 		}
 	}
+	replaceJSON = fmt.Sprintf("%v]", replaceJSON)
+	removeJSON = fmt.Sprintf("%v]", removeJSON)
 
-	return jsonPatch, nil
+
+	if priorReplace {
+		replacePatch, err = jsonpatch.DecodePatch([]byte(replaceJSON))
+		if err != nil {
+			return nil, err
+		}
+	}
+	if priorRemove {
+		removePatch, err = jsonpatch.DecodePatch([]byte(removeJSON))
+		if err != nil {
+			return nil, err
+		}
+	}
+	return append(replacePatch, removePatch...), nil
 }
 
-func UpdateServiceAccount(u unstructured.Unstructured) (jsonpatch.Patch, error) {
-	pullSecrets := getPullSecretReferencesServiceAccount(u)
+func updateSecret(
+	pullSecret *v1.LocalObjectReference,
+	secretPath string,
+	fields openshiftOptionalFields) (jsonpatch.Patch, error) {
+	var err error
 
-	jsonPatch := jsonpatch.Patch{}
+	patch := jsonpatch.Patch{}
+	var patchJSON string
 
-	check := u.GetName() + "-dockercfg-"
+	if pullSecret == nil || len(pullSecret.Name) == 0  {
+		return patch, nil
+	}
 
-	for n, secret := range pullSecrets {
-		if strings.HasPrefix(secret.Name, check) {
+	newSecret, ok := fields.PullSecretReplacement[pullSecret.Name]
+	// replacement found
+	if ok {
+		patchJSON = fmt.Sprintf(replaceSecretOp, secretPath+"/name", newSecret)
+	} else if fields.StripDefaultPullSecrets && isDefault(pullSecret.Name) {
+		patchJSON = fmt.Sprintf(removeSecretOp, secretPath)
+	}
 
-			patchJSON := fmt.Sprintf(`[
-{ "op": "remove", "path": "/imagePullSecrets/%v"}
-]`, n)
-			patch, err := jsonpatch.DecodePatch([]byte(patchJSON))
-			if err != nil {
-				return nil, err
-			}
-			jsonPatch = append(jsonPatch, patch...)
 
+	if len(patchJSON) > 0 {
+		patch, err = jsonpatch.DecodePatch([]byte(patchJSON))
+		if err != nil {
+			return nil, err
 		}
 	}
+	return patch, nil
+}
 
-	secrets := getSecretReferencesServiceAccount(u)
-	for n, secret := range secrets {
-		if strings.HasPrefix(secret.Name, check) {
+func nonInitialDelimiter(priorEntries bool) string {
+    if priorEntries {
+        return ","
+    } else {
+        return ""
+    }
+}
+func UpdateServiceAccount(u unstructured.Unstructured) (jsonpatch.Patch, error) {
+	jsonPatch := jsonpatch.Patch{}
+	check := u.GetName() + "-dockercfg-"
+	var err error
 
-			patchJSON := fmt.Sprintf(`[
-{ "op": "remove", "path": "/secrets/%v"}
-]`, n)
-			patch, err := jsonpatch.DecodePatch([]byte(patchJSON))
-			if err != nil {
-				return nil, err
-			}
-			jsonPatch = append(jsonPatch, patch...)
+	pullSecrets := getPullSecretReferencesServiceAccount(u)
+	pullSecretsPatch := jsonpatch.Patch{}
+	pullSecretsJSON := `[`
+	var priorPullSecret bool
+	for i := len(pullSecrets)-1; i >= 0; i-- {
```

Comment:
@sseago this is the one that I was thinking could use a comment as well as line 199